config/config.yml

```diff
@@ -8,4 +8,6 @@ MODEL_TYPE: 'llama'
 MODEL_BIN_PATH: 'models/leo-hessianai-7b-chat.Q5_K_M.gguf'
 MAX_NEW_TOKENS: 256
 TEMPERATURE: 0.01
-DEVICE: 'cpu'  # options: cpu, mps, gpu
\ No newline at end of file
+DEVICE: 'cpu'  # options: cpu, mps, gpu
+DOCUMENT_GENERATION: False # True for Document Generation, False for Document Questioning
+FACTCHECKING: True
```

Comment:
Ja, so kann man das machen, aber für Variablen, die je nach Umgebung unterschiedlich sind (aka Umgebungsvariablen) bietet es sich an, die Projektkonfiguration von der Umgebungskonfiguration zu trennen. Im Web- bzw. VM/Container-Kontext wird dabei oft auf Umgebungsvariablen zurückgegriffen, die entweder als Kommandozeilenargument gesetzt werden oder in einer `.env`-Datei. `.env`-Dateien werden nicht comitted und können deshalb auch secrets beinhalten. Mit [pythont-dotenv](https://github.com/theskumar/python-dotenv) kann man relativ einfach `.env` in Python nutzen.

Reply:
damit man nicht vergisst, welche Variablen in .env möglich sind, kann man mehrere Ansätze wählen. Zum Beispiel:

- Ansatz 1: Alles was in config.yml möglich ist kann via `.env` überschrieben werden.
- Ansatz 2: Eine `env.sample` anlegen mit einer kompletten Beispielkonfiguration, in der alle Parameter außerdem die Standardwerte anzeigen.

Reply:
Auch Gesprächsthema für Montag. Selber noch keine Erfahrung damit. Ich werde mich dazu einlesen.

Reply:
Eine weitere Lösung, die wir heute diskutieren können wäre [sops](https://github.com/getsops/sops) mit [age](https://github.com/FiloSottile/age) keys für jeden developer.